Initial version for .NET SDK 7.0.302

Xzelsius opened the pull request:
Same feature set as the .NET 6 images, but with .NET 7 SDK.

Xzelsius committed:
chore: add initial version

Xzelsius commented:
Pending tasks
- [x] https://github.com/swissgrc/.github/pull/48 needs to be merged
- [x] https://github.com/swissgrc/docker-azure-pipelines-dotnet/pull/115 needs to be merged
- [x] Set custom unstable tag name
- [x] Sonar Cloud Setup
- [x] Update this PR again

pascalberger commented on GitReleaseManager.yaml:
This file is no longer required

Xzelsius committed:
chore: remove obsolete file

Xzelsius committed:
refactor: use our own git base image

The new base image already provides docker cli and git.
Thus it simplifies the dockerfile.

Xzelsius committed:
chore: update to main publish-image workflow

Xzelsius committed:
refactor: remove unused stuff from dockerfile

Xzelsius committed:
refactor: adjust to latest publish-image workflow

Xzelsius committed:
docs: update README.md
